src/event-stream/event-server.ts

```diff
@@ -743,12 +745,27 @@ export async function startEventServer(opts: {
     }
   });
 
+  const handleBnsImport = async (blockMessage: CoreNodeBlockMessage) => {
+    const bnsDir = process.env.BNS_IMPORT_DIR;
+    if (blockMessage.block_height === 1 && bnsDir && process.env.TSV_IMPORT_IN_PROCESS !== 'true') {
+      const configState = await db.getConfigState();
+      if (!configState.bns_names_onchain_imported && !configState.bns_subdomains_imported) {
+        const bnsGenesisBlock = getBnsGenesisBlockFromBlockMessage(blockMessage);
+        logger.verbose('Starting V1 BNS names import');
+        await importV1BnsNames(db, bnsDir, bnsGenesisBlock);
+        logger.verbose('Starting V1 BNS subdomains import');
+        await importV1BnsSubdomains(db, bnsDir, bnsGenesisBlock);
+      }
+    }
+  };
+
   app.post(
     '/new_block',
     asyncHandler(async (req, res, next) => {
       try {
-        const msg: CoreNodeBlockMessage = req.body;
-        await messageHandler.handleBlockMessage(opts.chainId, msg, db);
+        const blockMessage: CoreNodeBlockMessage = req.body;
+        await messageHandler.handleBlockMessage(opts.chainId, blockMessage, db);
+        await handleBnsImport(blockMessage);
```

Comment:
Have you tested locally that this works as expected? The code looks correct but I worry that because this BNS import process will take a long time (~5 minutes), the Stacks node will be left waiting for a response from the API acknowledging the block message (because you're `await`ing here) and decide to declare a timeout, dropping the message or worse trying again (which would re-trigger the same functionality).

@zone117x maybe has more context here on the core node behavior.

Maybe a better idea would be **not** to `await` here and just let the promise run in parallel, but we'd need to verify if this doesn't cause any problems.

Reply:
That sounds right to me. We can make bnsimport middleware that runs after the /new_block handler. That way the req will be handled the same as before this change, sending the client a response as soon as the handleblockmessage completes.

Reply:
Hmm, I'm not sure what's more error prone: A) blocking the event-observer response for longer than usual, or B) running the BNS insertion sql at the same time as later new blocks are being ingested with its sensitive sql code.

With either approach, we should test that the BNS import completes and the first few blocks are ingested while using the mainnet dataset. This could be tested locally fairly quickly. @BLuEScioN is this something you can test to confirm nothing breaks?